MaterialsGalaxy/neutron_galaxy
# Add gsas datatypes

This PR will add GSAS datatypes: gpx, prm, instprm, raw and gsa. Sniffers are included for prm, instprm and gsa filetypes. Datatypes are being added to fix a bug in the GSASII refinement tools where the incorrect .dat datatype was provided as default to the tool python script causing an error:
```
Traceback (most recent call last):
  File "/opt/conda/envs/gsas2/lib/python3.10/runpy.py", line 196, in _run_module_as_main
    return _run_code(code, main_globals, None,
  File "/opt/conda/envs/gsas2/lib/python3.10/runpy.py", line 86, in _run_code
    exec(code, run_globals)
  File "/opt/conda/envs/gsas2/lib/python3.10/site-packages/gsas2_refinement/gsas2_refinement.py", line 33, in <module>
    run_gsas2_fit(
  File "/opt/conda/envs/gsas2/lib/python3.10/site-packages/gsas2_refinement/run_gsas2_fit.py", line 78, in run_gsas2_fit
    hist1 = gpx.add_powder_histogram(gsa_fn, prm_fn, fmthint="GSAS powder", databank=bank, instbank=bank)
  File "/app/GSASIIscriptable.py", line 2162, in add_powder_histogram
    histname, new_names, pwdrdata = load_pwd_from_reader(
  File "/app/GSASIIscriptable.py", line 1718, in load_pwd_from_reader
    Iparm1, Iparm2 = load_iprms(instprm, reader, bank=bank)
  File "/app/GSASIIscriptable.py", line 1675, in load_iprms
    raise ValueError('Expected .prm file, found: ', instfile)
ValueError: ('Expected .prm file, found: ', '/mnt/nfs/data/9/5/a/dataset_95a9a568-d94a-4c89-a08c-3f5ddc5c01af.dat')
```
This error can currently be recreated by using the simple refinement tool in the galaxy-dev server with any files matching the tool inputs which together would result in a successful refinement in the desktop GSAS application. 
 
## How to test the changes?
(Select all options that apply)
  1. Upload the listed filetypes in a galaxy instance and check if the sniffer detects the correct type
  2. check that the list of types also includes gpx and raw.
  3. run the local gsas refinement tool (if available) and ensure the output files are corrected identified
  4. follow the tests described in the doc strings for the sniffing functions in datatypes/gsas2.py 

## License
